### PR TITLE
Feat/project postings

### PR DESCRIPTION
**Pull request by stalhaali**
Implemented:

- Filter project by
   - Location
   - Number of members
   - Category
- Project post thumbnail includes title, location, number of members, and creator.
- Clicking on project leads to the projects information page.
- API route `GET /api/v1/projects/postings` (paginated)

### CODE REVIEW

**Review by TianYueDong — APPROVED** [review]
LGTM

File: Buddies.API/Controllers/ProjectsController.cs [hunk 1]
```diff
@@ -25,6 +25,79 @@ public ProjectsController(ApiContext context, UserManager<User> userManager)
...
+
+        /// <summary>
+        /// API route GET /api/v1/postings/{filters}/... for fetching all projects
+        /// that pass the filters.
+        /// </summary>
+        [HttpGet("postings/{location}/{members}/{category}/{page}/{results}")]
+        public async Task<ActionResult> GetProjectListing(string location, int members, string category, int page, float results)
+        {
+
+            // var projectList = await _context.Projects.ToListAsync();
```
Comment (andy9775) [inline Buddies.API/Controllers/ProjectsController.cs]:
```suggestion
```

File: Buddies.Web/src/components/Websites.tsx [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 import Typography from '@mui/material/Typography';
 import CustomizedDialogs from './dialog';
 
-const Websites = ({ logCheck }:{ logCheck: boolean | null }) => {
+interface Props {
+  /**
+   * variable to track whether a user is logged in
+   * and viewing his profile
+   */
+  logCheck: boolean | null
```
Comment (andy9775) [inline Buddies.Web/src/components/Websites.tsx]: reword this? `isViewingOwnProfile`?

**Review by andy9775 — APPROVED** [review]
Some minor nits otherwise LGTM

**Pull request merged**